pkgs/servers/search/opensearch/opensearch-home-fix.patch

```diff
@@ -0,0 +1,26 @@
+diff -Naur a/bin/opensearch-env b/bin/opensearch-env
+--- a/bin/opensearch-env	2017-12-12 13:31:51.000000000 +0100
++++ b/bin/opensearch-env	2017-12-18 19:51:12.282809695 +0100
+@@ -19,18 +19,10 @@
+   fi
+ done
+ 
+-# determine OpenSearch home; to do this, we strip from the path until we find
+-# bin, and then strip bin (there is an assumption here that there is no nested
+-# directory under bin also named bin)
+-OPENSEARCH_HOME=`dirname "$SCRIPT"`
+-
+-# now make OPENSEARCH_HOME absolute
+-OPENSEARCH_HOME=`cd "$OPENSEARCH_HOME"; pwd`
+-
+-while [ "`basename "$OPENSEARCH_HOME"`" != "bin" ]; do
+-  OPENSEARCH_HOME=`dirname "$OPENSEARCH_HOME"`
+-done
+-OPENSEARCH_HOME=`dirname "$OPENSEARCH_HOME"`
++if [ -z "$OPENSEARCH_HOME" ]; then
++    echo "You must set the OPENSEARCH_HOME var" >&2
```

Comment:
```
❯ OPENSEARCH_HOME=/tmp/opensearch ./result/bin/opensearch-node
could not find java in bundled jdk at /tmp/opensearch/jdk/bin/java
```

So what it actually requires is essentially like a `JAVA_HOME`? But I cannot just pass in for example `openjdk.out`, because it has `bin/java`, but not `jdk/bin/java`.

```
❯ OPENSEARCH_HOME=/nix/store/ch095in4hr481iqv11s55ks9ksl2c49i-openjdk-19.0.1+10/ ./result/bin/opensearch-node
could not find java in bundled jdk at /nix/store/ch095in4hr481iqv11s55ks9ksl2c49i-openjdk-19.0.1+10//jdk/bin/java
```

So then I symlink myself something together, and now I'm stuck here:

````
❯ OPENSEARCH_HOME=/tmp/opensearch ./result/bin/opensearch
Error: Could not find or load main class org.opensearch.tools.java_version_checker.JavaVersionChecker
Caused by: java.lang.ClassNotFoundException: org.opensearch.tools.java_version_checker.JavaVersionChecker
````

What does it take to get this running? I think we really need a small module to encapsulate this requirement.

Reply:
see this as example how to get it up running :)
https://github.com/cachix/devenv/blob/main/src/modules/services/elasticsearch.nix#L35-L80

Reply:
If they haven't diverge enough already, a compatible module would be amazing.

Reply:
Make the existing elastic search one compatible or a new one?

Reply:
That really depends on how compatible they still are.

Reply:
I added a module `services.opensearch.enable = true` and works

![image](https://user-images.githubusercontent.com/6224096/215871251-866fd6ef-024e-4744-b821-fea8ea5f41b1.png)

Reply:
Cool, what do you think about adding a small nixos test to continuously validate that everything is still working?

Reply:
I will look tomorrow at the moment I don't have a machine with kvm enabled

Reply:
test is added :)